src/arkode/arkode_lsrkstep_io.c

```diff
@@ -273,81 +273,93 @@ int LSRKStepSetSSPStageNum(void* arkode_mem, int num_of_stages)
   retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
                                         &step_mem);
-  if (retval != ARK_SUCCESS) { return (retval); }
+  if (retval != ARK_SUCCESS) { return retval; }
 
-  switch (step_mem->LSRKmethod)
+  if (!step_mem->is_SSP)
   {
-  case ARKODE_LSRK_SSP_S_2:
-    if (num_of_stages < 2)
-    {
-      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
-                      "num_of_stages must be greater than or equal to 2");
-      return (ARK_ILL_INPUT);
-    }
-    break;
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Call this function only for SSP methods: Use "
+                    "LSRKStepSetMethod to declare SSP method type first!");
+    return ARK_ILL_INPUT;
+  }
 
-  case ARKODE_LSRK_SSP_S_3:
-    if (num_of_stages < 9 || (SUNRceil(SUNRsqrt(num_of_stages)) !=
-                              SUNRfloor(SUNRsqrt(num_of_stages))))
+  if (num_of_stages <= 0)
+  {
+    switch (step_mem->LSRKmethod)
     {
+    case ARKODE_LSRK_SSP_S_2: step_mem->req_stages = 10; break;
+
+    case ARKODE_LSRK_SSP_S_3: step_mem->req_stages = 9; break;
+
+    case ARKODE_LSRK_SSP_10_4: step_mem->req_stages = 10; break;
+
+    default:
       arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__,
-                      __FILE__, "num_of_stages must be a perfect square greater than or equal to 9");
-      return (ARK_ILL_INPUT);
+                      __FILE__, "Call LSRKStepSetMethod to declare SSP method type first!");
+      return ARK_ILL_INPUT;
+      break;
     }
-    break;
-
-  case ARKODE_LSRK_SSP_10_4:
-    if (num_of_stages != 10)
+    return ARK_SUCCESS;
+  }
+  else
+  {
+    switch (step_mem->LSRKmethod)
     {
-      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
-                      "SSP10_4 method has a prefixed num_of_stages = 10");
-      return (ARK_ILL_INPUT);
+    case ARKODE_LSRK_SSP_S_2:
+      if (num_of_stages < 2)
+      {
+        arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                        "num_of_stages must be greater than or equal to 2, or "
+                        "set it less than or equal to 0 to reset the default "
+                        "value");
+        return ARK_ILL_INPUT;
+      }
+      break;
+
+    case ARKODE_LSRK_SSP_S_3:
+      if (num_of_stages < 4 || ((int)SUNIceil(SUNRsqrt(num_of_stages)) !=
+                                (int)SUNIfloor(SUNRsqrt(num_of_stages))))
+      {
+        arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                        "num_of_stages must be a perfect square greater than "
+                        "or equal to 4, or set it less than or equal to 0 to "
+                        "reset the default value");
+        return ARK_ILL_INPUT;
+      }
+      if (num_of_stages == 4) { ark_mem->step = lsrkStep_TakeStepSSP43; }
```

Comment:
So SSP43 is a special case of SSPs3? Add a comment to the block above the `SSPs3` and `SSP43` take step functions to clarify.